### PR TITLE
Reader: Auto-expand comment textarea

### LINKED ISSUE #695: Reader: expand comments textarea as user types
The single-line comment textarea is too small. It should expand to fit the content entered by the user.

screen shot 2015-11-24 at tue nov 24 5 28 37 pm
<sup>Seen at https://wordpress.com/read/post/id/92384102/110 using Safari iOS 8.4.1 on iPhone 6</sup>

From bluefuton:

<blockquote>We propose to use:

https://github.com/andreypopp/react-textarea-autosize</blockquote>

<blockquote>https://github.com/andreypopp/react-textarea-autosize does not seem to support assigning a ref, which all of our events hang off.

gcorne suggested that instead: "you might look at the tinymce component and the autosize npm package that it uses".</blockquote>

### PR DESCRIPTION
**Pull request by shaunandrews**
comment autoexpand

Adding a really basic way of expanding the comment textarea based on the scrollHeight. This makes it easier to write longer comments and aims to fix #695.

### CODE REVIEW

File: client/reader/comments/form.jsx [hunk 1]
```diff
@@ -43,6 +43,14 @@ var PostCommentForm = React.createClass( {
 			event.preventDefault();
 			this.submit();
 		}
+
+		// Adjust the height of the textarea as needed.
+		var commentTextNode = this.refs.commentText;
+
+		setTimeout( function() {
+			commentTextNode.style.cssText = 'height: auto';
+			commentTextNode.style.cssText = 'height:' + commentTextNode.scrollHeight + 'px';
```
Comment (blowery) [inline client/reader/comments/form.jsx]: i think you can just set style.height instead of going through cssText

Comment (shaunandrews) [inline client/reader/comments/form.jsx]: Ah, nice. Switched to `style.height` and works great.

**Comment by blowery** [comment]
eeek. when you get near the bottom of the screen it goes bonkers on mac'n'chrome. I think I might have a fix.

woof

**Comment by blowery** [comment]
Hrm, layout is funky on Firefox

screenshot

**Comment by blowery** [comment]
alignment is a tick off in IE11

screenshot

**Comment by shaunandrews** [comment]
Notifications (which also auto-expands the input) sticks the `SEND` label to the top:

screen shot 2016-01-21 at 2 06 02 pm

@drw158 What do you think? Top or bottom?

**Comment by blowery** [comment]
Here's a vid of it stuck bottom

stuck-bottom

**Comment by shaunandrews** [comment]
I prefer it on the bottom, especially when you consider that really long comments could (and probably will) scroll the `SEND` label out of view if its at the top.

**Comment by blowery** [comment]
Alright, back to the top because `bottom` is just too weird. Every browser is sticking the bottom of the container in a different place, so it was nearly impossible to get it all to line up.

Looks good to me on all browsers. @shaunandrews , can you check mobile?

**Comment by drw158** [comment]
Bottom could be nice, but I think top works as well too.

**Comment by shaunandrews** [comment]
Pushed a fix for iOS Safari. Only other issue I noticed was Chrome on Android (Nexus 5): Tapping the textarea opens the keyboard, but doesn't scroll down, so the textarea is _hidden_ under the keyboard.

screenshot_2016-01-21-15-08-00

**Comment by blowery** [comment]
I think we're good to go. We can address the Nexus issues in a separate PR.

**Pull request merged**